sdk/cosmos/azure-cosmos/test/test_auto_scale.py

```diff
@@ -0,0 +1,177 @@
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import unittest
+import azure.cosmos.cosmos_client as cosmos_client
+import azure.cosmos.exceptions as exceptions
+from azure.cosmos import Offer, PartitionKey, http_constants
+import pytest
+import test_config
+
+pytestmark = pytest.mark.cosmosEmulator
+
+
+@pytest.mark.usefixtures("teardown")
+class AutoScaleTest(unittest.TestCase):
+    host = test_config._test_config.host
+    masterKey = test_config._test_config.masterKey
+    connectionPolicy = test_config._test_config.connectionPolicy
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
+                cls.host == '[YOUR_ENDPOINT_HERE]'):
+            raise Exception(
+                "You must specify your Azure Cosmos account values for "
+                "'masterKey' and 'host' at the top of this class to run the "
+                "tests.")
+
+        cls.client = cosmos_client.CosmosClient(cls.host, cls.masterKey, consistency_level="Session",
+                                                connection_policy=cls.connectionPolicy)
+        cls.created_database = cls.client.create_database(test_config._test_config.TEST_DATABASE_ID)
+
+    def test_auto_scale_max_throughput(self):
+        created_container = self.created_database.create_container(
+            id='container_with_auto_scale_settings',
+            partition_key=PartitionKey(path="/id"),
+            offer_throughput=Offer(auto_scale_max_throughput=5000, auto_scale_increment_percentage=0)
+
+        )
+        created_container_properties = created_container.get_throughput()
+        # Testing the input value of the max_throughput
+        self.assertEqual(
+            created_container_properties.properties['content']['offerAutopilotSettings']['maxThroughput'], 5000)
+
+        self.created_database.delete_container(created_container)
+
+        # Testing the incorrect passing of an input value of the max_throughput to verify negative behavior
+        try:
+            created_container = self.created_database.create_container(
+                id='container_with_wrong_auto_scale_settings',
+                partition_key=PartitionKey(path="/id"),
+                offer_throughput=Offer(auto_scale_max_throughput=200, auto_scale_increment_percentage=0)
+
+            )
+        except exceptions.CosmosHttpResponseError as e:
+            self.assertEqual(e.status_code, http_constants.StatusCodes.BAD_REQUEST)
```

Comment:
Do we get any error message for an incorrect percentage value?
It would be pretty hard to debug the problem if all I got was "bad request"

Reply:
Would you like me to print the error message there?

Reply:
When asserting with the message rather than the status code we get assertion errors, so what should I do with that?